### PR TITLE
Update release notes labeling

### PR DESCRIPTION
**Pull request by carolineechen**
- fix retrieve PR script to handle commits with unrecognized/invalid PR numbers, such as in 7b6b2d000023e2aa3365b769866c5f375e0d5fda
- add modifications similar to pytorch's [#71917](https://github.com/pytorch/pytorch/pull/71917), [#72085](https://github.com/pytorch/pytorch/pull/72085)

### CODE REVIEW

**Review by mthrok — APPROVED** [review]
Stamp

**Review by mthrok — COMMENTED** [review]
I am curious, why not making it a CI job and simply make it merge blocking job?

PyTorch core needs more flexibility because of frequent external contributors, but we do not have one.

File: .github/process_commit.py [hunk 1]
```diff
@@ -36,15 +37,18 @@
     "perf",
     "other",
 }
+GITHUB_TOKEN = os.environ.get("GITHUB_TOKEN")
```
Comment (mthrok) [inline .github/process_commit.py:40]: Is a maintainer expected to run this script manually and locally?
If so, I think it's better to 
- fail if the env var is not defined, and
- provide `--help` option and the `--help` option to work even when GITHUB_TOKEN is not set.

Comment (carolineechen) [reply .github/process_commit.py:40]: this script shouldn't be run by maintainers, but by the github workflow when there is a push to main. the github token is set [here](https://github.com/pytorch/audio/blob/e8f7e192d4b47f0ec25b4920bd29ffe216739ebe/.github/workflows/pr-labels.yml#L26)

**Comment by carolineechen** [comment]
> I am curious, why not making it a CI job and simply make it merge blocking job?

Wanted to try resolving the current approach if changes were simple, as most of the scripts and setup are already implemented. I'm not as familiar with implementing the CI job yet, but if this is still not working or if the team prefers (or if you'd like to make the change), happy to use a CI job to replace this instead.

**Review by nateanl — APPROVED** [review]

**Pull request closed without merging**